griptape/tools/rest_api_client/tool.py

```diff
@@ -185,19 +193,20 @@ def get(self, params: dict) -> BaseArtifact:
             ),
         }
     )
-    def delete(self, params: dict = None) -> BaseArtifact:
+    def delete(self, params: dict) -> BaseArtifact:
         from requests import delete, exceptions
 
         values = params["values"]
         base_url = self.base_url
-        path = self.path
-
+        path = values.get("path", self.path)
```

Comment:
`path` is not in the schema so it will not ever be present in `values`. What is the use case you have in mind here?

Reply:
discussed here: https://github.com/griptape-ai/griptape/pull/535#discussion_r1442218071

Reply:
Same point here, I think it seems risky to have the LLM take guesses at the `path`. I do think we should have better methods for creating a single `RestApiClient` for multiple paths, but that feels out of scope for this PR.

Reply:
We've experimentally shown, under specific circumstances, that an LLM can build the entire request including the path and schema (and perhaps headers, too), but it's a very specific case and takes quite a few prerequisites and some massaging to get it to work right. I agree with @collindutter here that it's risky for the default behavior of the tool to include sending authenticated `DELETE` requests at guessed paths and that those should be deterministic configurations.

Reply:
@andrewfrench  & @collindutter thank you, totally understand the concerns. will update the pr this weekend and ping you when the changes are ready.

Reply:
@andrewfrench @collindutter  ping!  :D